File: Kap13/JavaScript/Listing39_47/mygraphs/FlaechenElm.mjs
import GrafikElm from "./GrafikElm.mjs";

export default class FlaechenElm extends GrafikElm {
    flaeche = 0;
     
    constructor(initRandfarbe, initFuellfarbe) {
        super(initRandfarbe);
        if (new.target === FlaechenElm) {
            throw TypeError("Error!");
        }
        this.fuellfarbe = initFuellfarbe;
        console.log("FlaechenElm erzeugt.");
    }
   
    berechneFlaeche() {
        throw TypeError("Error!");
    }
  
    getFuellfarbe() {
      return this.fuellfarbe;
    }

    setFuellfarbe(newFuellfarbe) {
      this.fuellfarbe = newFuellfarbe;
    }

    getFlaeche() {
      return this.flaeche;
    }

    printInfo() {
      super.printInfo();
      console.log("Fuellfarbe: " + this.fuellfarbe);
      console.log("Flaeche: " + this.flaeche);
    }
}

File: Kap13/Java/Listing20_21/mygraphs/FlaechenElm.java
package Kap13.Java.Listing20_21.mygraphs;

public class FlaechenElm extends GrafikElm {
    protected int fuellfarbe;
    protected double flaeche = 0;
     
    public FlaechenElm(int initRandfarbe, int initFuellfarbe) {
      super(initRandfarbe);
      fuellfarbe = initFuellfarbe;
      System.out.println("FlaechenElm erzeugt.");
    }
    public int getFuellfarbe() {
      return fuellfarbe;
    }
    public void setFuellfarbe(int newFuellfarbe) {
      fuellfarbe = newFuellfarbe;
    }
    public double getFlaeche() {
      return flaeche;
    }
    public void printInfo() {
      super.printInfo();
      System.out.println("Fuellfarbe: " + fuellfarbe);
      System.out.println("Flaeche: " + flaeche);
    }
  }	
  
  
